projects/plugins/social/src/class-jetpack-social.php

```diff
@@ -127,4 +132,41 @@ public function plugin_settings_page() {
 			<div id="jetpack-social-root"></div>
 		<?php
 	}
+
+	/**
+	 * Whitelist the `active_modules` option for Jetpack Sync.
+	 *
+	 * If the Jetpack plugin is active, this filter won't run (because it has
+	 * a lower priority and checks for the existence of `active_modules`).
+	 *
+	 * @param array $callables Array of callables to sync.
+	 * @return array
+	 */
+	public function filter_sync_callable_whitelist( $callables ) {
+		$callables['active_modules'] = function () use ( $callables ) {
+			$synced_active_modules = array_key_exists( 'active_modules', $callables ) ? $callables['active_modules']() : array();
```

Comment:
I wonder if we're doing something wrong here, by initialising this to an empty array. I have 12 modules active locally, but when the option is sync'd with WPCOM it only has `publicize`. Should this get initialised to the current value of `jetpack_active_modules`?

Reply:
This is intentional behavior—we don't want these modules to appear as active when the Jetpack plugin isn't active. When the Jetpack plugin is active, this filter doesn't run and all active modules get synced from there instead.

Reply:
Does it definitely not run?

> we don't want these modules to appear as active when the Jetpack plugin isn't active

Right, that makes more sense, but what about other standalone plugins? Is the only way to get the `active_modules` option to sync, using this callback method? If so, that's fine, because we should be preserving anything added by other plugins, but we can't make the assumption that only the Jetpack plugin provides the other modules' functionality.

Reply:
> Does it definitely not run?

I misspoke here, it _does_ run with the Jetpack plugin active as well. Originally, the implementation was to not run this filter when another plugin was adding `active_modules` to the whitelist. However, as you correctly pointed out, another standalone plugin might be syncing just their own module (e.g. `['search']` or something similar), in which case Publicize wouldn't get added.

In the current implementation, any existing filters are respected (either from the Jetpack plugin or another standalone plugin). Publicize simply gets appended to this list when Jetpack Social is active (and the Publicize module is active).